### PR TITLE
Switch to Github Actions

### PR DESCRIPTION
**Pull request by Korbeil**

### CODE REVIEW

**Review by welcoMattic — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: CI
+on: [push, pull_request]
```
Comment (lyrixx) [inline .github/workflows/ci.yaml:2]: * Could you add a cron too, like https://github.com/jolicode/docker-starter/blob/main/.github/workflows/ci.yml#L8-L10
* we will only build push on master and pr: https://github.com/jolicode/docker-starter/blob/main/.github/workflows/ci.yml#L4-L7

(I contribute to this lib via this repo and not my fork. So there is no need to trigger the workflow twice)

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 15
```
Comment (lyrixx) [inline .github/workflows/ci.yaml:7]: can be removed

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        php-versions:  ['7.4', '8.0']
+    name: tests-${{ matrix.php-versions }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@master
+      - name: cache-composer
+        uses: actions/cache@v2
+        with:
+          path: ~/.composer/cache/
+          key: composer-${{ matrix.php-versions }}-${{ github.sha }}
```
Comment (lyrixx) [inline .github/workflows/ci.yaml:18]: can be removed

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 15
+      matrix:
+        php-versions:  ['7.4', '8.0']
+    name: tests-${{ matrix.php-versions }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@master
```
Comment (lyrixx) [inline .github/workflows/ci.yaml:13]: This is fragile
```suggestion
        uses: actions/checkout@v2
```

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  tests-lowest:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@master
+      - name: cache-composer-lowest
+        uses: actions/cache@v2
+        with:
+          path: ~/.composer/cache/
+          key: composer-lowest-${{ github.sha }}
```
Comment (lyrixx) [inline .github/workflows/ci.yaml:40]: Can be removed

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+          php-version: ${{ matrix.php-versions }}
+      - name: composer install
+        run: composer update --prefer-stable
+      - name: Build docker
+        run: docker build -t httpbin .
+      - name: Run docker
+        run: docker run -d -p 8081:80 httpbin
+      - name: tests
+        run: composer test
+  tests-lowest:
```
Comment (lyrixx) [inline .github/workflows/ci.yaml:31]: Can it be "merged" with the main job?
see https://github.com/jolicode/JoliNotif/blob/main/.github/workflows/ci.yml#L23-L35 for an example

**Review by lyrixx — COMMENTED** [review]
Super cool, thanks 💛 
I left some comments, could you look at them, please?

**Pull request merged**